From e0ec8488151be3b5c38e1af5e561e8c14046950c Mon Sep 17 00:00:00 2001
From: Thitat Auareesuksakul <thitat@flux.ci>
Date: Thu, 25 Aug 2022 03:03:40 +0700
Subject: [PATCH 4/4] scripts/extract-ikconfig: add zstd compression support

Add extract-ikconfig support for kernel images compressed with zstd.

Signed-off-by: Thitat Auareesuksakul <thitat@flux.ci>
Tested-by: Nicolas Schier <n.schier@avm.de>
Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 scripts/extract-ikconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/extract-ikconfig b/scripts/extract-ikconfig
index 3b42f255e..8df33e7d6 100755
--- a/scripts/extract-ikconfig
+++ b/scripts/extract-ikconfig
@@ -62,6 +62,7 @@ try_decompress 'BZh'          xy    bunzip2
 try_decompress '\135\0\0\0'   xxx   unlzma
 try_decompress '\211\114\132' xy    'lzop -d'
 try_decompress '\002\041\114\030' xyy 'lz4 -d -l'
+try_decompress '\050\265\057\375' xxx unzstd
 
 # Bail out:
 echo "$me: Cannot find kernel config." >&2
-- 
2.37.3.485.gbe1a02a17e

